src/black/__init__.py

```diff
@@ -1103,6 +1106,11 @@ def _format_str_once(src_contents: str, *, mode: Mode) -> str:
     dst_contents = []
     for block in dst_blocks:
         dst_contents.extend(block.all_lines())
+    if mode.preview and not dst_contents:
+        normalized_content, _, newline = decode_bytes(src_contents.encode("utf-8"))
```

Comment:
This feels a bit janky to using `decode_bytes` like this, but I can't think of anything better. Could we add a comment to this whole logic explaining what it does?